opamp/observiq/agent_config_manager.go

```diff
@@ -0,0 +1,167 @@
+// Copyright  observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package observiq
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/observiq/observiq-otel-collector/opamp"
+	"github.com/open-telemetry/opamp-go/protobufs"
+	"go.uber.org/zap"
+)
+
+const (
+	// CollectorConfigName is the key of the collector config in OpAmp
+	CollectorConfigName = "collector.yaml"
+	// ManagerConfigName is the key of the manager config in OpAmp
+	ManagerConfigName = "manager.yaml"
+	// LoggingConfigName is the key of the logging config in OpAmp
+	LoggingConfigName = "logging.yaml"
+)
+
+// acceptableConfigs is a lookup of configs that are able to be written/updated
+var acceptableConfigs = map[string]struct{}{
+	CollectorConfigName: {},
+	ManagerConfigName:   {},
+	LoggingConfigName:   {},
+}
+
+// Enforce interface
+var _ opamp.ConfigManager = (*AgentConfigManager)(nil)
+
+// AgentConfigManager keeps track of active configs for the agent
+type AgentConfigManager struct {
+	configMap map[string]*opamp.ManagedConfig
+	logger    *zap.SugaredLogger
+}
+
+// NewAgentConfigManager creates a new AgentConfigManager
+func NewAgentConfigManager(defaultLogger *zap.SugaredLogger) *AgentConfigManager {
+	return &AgentConfigManager{
+		configMap: make(map[string]*opamp.ManagedConfig),
+		logger:    defaultLogger.Named("config manager"),
+	}
+}
+
+// AddConfig adds a config to be tracked by the config manager.
+// If the config already is tracked it'll be overwritten with the new managed config
+func (a *AgentConfigManager) AddConfig(configName string, managedConfig *opamp.ManagedConfig) {
+	a.configMap[configName] = managedConfig
+}
+
+// ComposeEffectiveConfig reads in all config files and calculates the effective config
+func (a *AgentConfigManager) ComposeEffectiveConfig() (*protobufs.EffectiveConfig, error) {
+	contentMap := make(map[string]*protobufs.AgentConfigFile, len(a.configMap))
+
+	// Used to track config names to alphabetize later in hash compution
+	configNames := make([]string, 0, len(a.configMap))
```

Comment:
is this actually used or can we ditch this?

Reply:
Nope artifact of a previous implementation. Good catch.